--- a/vanilla_files/1.8.7/gui/market_panel.gui
+++ b/gui/market_panel.gui
@@ -32,10 +32,52 @@
 		
 		datacontext = "[MarketPanel.GetMarket]"
 		
-		blockoverride "window_header_name"
-		{
-			raw_text = "#BOLD [Market.GetOwner.GetFlagTextIcon] [Market.GetName]#!"
+		### BEGIN MODULAR UI PATCH
+		blockoverride "window_header" {
+			default_block_window_header = {
+				blockoverride "window_header_name" {
+					raw_text = "#BOLD [Market.GetOwner.GetFlagTextIcon] [Market.GetName]#!"
+				}
+				blockoverride "top_left_button" {
+					# extension hook
+					market_panel_header_top_left_button = {
+						position = { 48 16 }
+						size = { 30 30 }
+					}
+				}
+				blockoverride "bottom_left_button" {
+					# extension hook
+					market_panel_header_bottom_left_button = {
+						position = { 48 59 }
+						size = { 30 30 }
+					}
+				}
+				blockoverride "pin_visibility" {
+					visible = yes
+				}
+				blockoverride "pin_properties" {
+					visible = "[MarketPanel.AccessMarket.IsPinnedInOutliner]"
+					onclick = "[MarketPanel.AccessMarket.TogglePinInOutliner]"
+					datacontext = "[MarketPanel.AccessMarket]"
+					tooltip = "UNPIN_MARKET"
+				}
+				blockoverride "unpin_properties" {
+					visible = "[Not(MarketPanel.AccessMarket.IsPinnedInOutliner)]"
+					onclick = "[MarketPanel.AccessMarket.TogglePinInOutliner]"
+					datacontext = "[MarketPanel.AccessMarket]"
+					tooltip = "PIN_MARKET"
+				}
+				blockoverride "goto_visibility" {
+					visible = yes
+				}
+				blockoverride "goto_properties" {
+					onclick = "[Market.AccessMarketCapital.ZoomToCapital]"
+					tooltip = "ZOOM_TO_MARKET_CAPITAL"
+					enabled = "[Not( Market.AccessMarketCapital.IsZoomingToCapital )]"
+				}
+			}
 		}
+		### END MODULAR UI PATCH
 		
 		blockoverride "fixed_top"
 		{
@@ -634,25 +676,43 @@
 			container = {
 				parentanchor = hcenter
 				
-				market_panel_details_content = {
+				### BEGIN MODULAR UI PATCH
+				container = {
 					visible = "[InformationPanel.IsTabSelected('default')]"
-					using = default_fade_in_out
+					market_panel_details_content_override = {}
+					market_panel_details_content = {
+						visible = "[EqualTo_int32(PdxGuiWidget.AccessParent.GetChildrenCount, '(int32)1')]"
+						using = default_fade_in_out
+					}
 				}
 
-				market_panel_trade_routes_content = {
+				container = {
 					visible = "[InformationPanel.IsTabSelected('trade_routes')]"
-					using = default_fade_in_out
+					market_panel_trade_routes_content_override = {}
+					market_panel_trade_routes_content = {
+						visible = "[EqualTo_int32(PdxGuiWidget.AccessParent.GetChildrenCount, '(int32)1')]"
+						using = default_fade_in_out
+					}
 				}
 
-				market_panel_food_security_content = {
+				container = {
 					visible = "[InformationPanel.IsTabSelected('food_security')]"
-					using = default_fade_in_out
+					market_panel_food_security_content_override = {}
+					market_panel_food_security_content = {
+						visible = "[EqualTo_int32(PdxGuiWidget.AccessParent.GetChildrenCount, '(int32)1')]"
+						using = default_fade_in_out
+					}
 				}
 
-				market_panel_states_content = {
+				container = {
 					visible = "[InformationPanel.IsTabSelected('states')]"
-					using = default_fade_in_out
+					market_panel_states_content_override = {}
+					market_panel_states_content = {
+						visible = "[EqualTo_int32(PdxGuiWidget.AccessParent.GetChildrenCount, '(int32)1')]"
+						using = default_fade_in_out
+					}
 				}
+				### END MODULAR UI PATCH
 			}
 		}
 
